operator/handler/config.go

```diff
@@ -39,16 +39,30 @@ const (
 	sidecarAnnotation = "tailing-sidecar"
 )
 
-// getConfigs parses configurations from annotation and joins them with configurations from TailingSidecars
-func getConfigs(annotations map[string]string, sidecarConfigs map[string]tailingsidecarv1.SidecarConfig) []tailingsidecarv1.SidecarConfig {
-	configs := make([]tailingsidecarv1.SidecarConfig, 0)
-
+// getConfigs gets configurations from TailingSidecars and annotations
+func getConfigs(annotations map[string]string, tailingSidecars []tailingsidecarv1.TailingSidecar) []tailingsidecarv1.SidecarConfig {
 	annotation, ok := annotations[sidecarAnnotation]
 	if !ok {
-		return configs
+		return []tailingsidecarv1.SidecarConfig{}
+	}
+
+	if annotation == "" {
+		handlerLog.Info("Empty tailing-sidecar annotation",
+			"annotation", annotation)
+		return []tailingsidecarv1.SidecarConfig{}
 	}
 
+	sidecarConfigs := joinTailingSidecarConfigs(tailingSidecars)
+
+	configs := parseAnnotation(annotation, sidecarConfigs)
+	return configs
+}
+
+// parseAnnotation parses configurations from 'tailing-sidecar' annotation and joins them with configurations from TailingSidecars
+func parseAnnotation(annotation string, sidecarConfigs map[string]tailingsidecarv1.SidecarConfig) []tailingsidecarv1.SidecarConfig {
```

Comment:
Food for thought: make an `Annotation` type and accept it instead of a `string` :) An idea for later perhaps.